source/fem-dem/gls_sharp_navier_stokes.cc

```diff
@@ -3606,6 +3607,62 @@ GLSSharpNavierStokesSolver<dim>::load_particles_from_file()
               shape_argument[2] = restart_data["shape_argument_2"][p_i];
               particles[p_i].initialize_shape("death star", shape_argument);
             }
+          else if (restart_data["type"][p_i] ==
+                   Shape<dim>::ShapeType::rbf_shape)
+            {
+              std::shared_ptr<RBFShape<dim>> shape =
+                std::dynamic_pointer_cast<RBFShape<dim>>(
+                  this->simulation_parameters.particlesParameters
+                    ->particles[p_i]
+                    .shape);
+
+              std::vector<double> shape_arguments;
+
+              size_t              number_of_nodes = shape->weights.size();
+              std::vector<double> weight;
+              std::vector<double> basis_function;
+              std::vector<double> support_radius;
+              std::vector<double> nodes_x;
+              std::vector<double> nodes_y;
+              std::vector<double> nodes_z;
+              weight.resize(number_of_nodes);
+              basis_function.resize(number_of_nodes);
+              support_radius.resize(number_of_nodes);
+              nodes_x.resize(number_of_nodes);
+              nodes_y.resize(number_of_nodes);
+              nodes_z.resize(number_of_nodes);
+
+              for (size_t n_i = 0; n_i < number_of_nodes; n_i++)
+                {
+                  weight[n_i]         = shape->weights[n_i];
+                  support_radius[n_i] = shape->support_radii[n_i];
+                  basis_function[n_i] =
+                    static_cast<double>(shape->basis_functions[n_i]);
+                  nodes_x[n_i] = shape->nodes[n_i][0];
+                  nodes_y[n_i] = shape->nodes[n_i][1];
+                  nodes_z[n_i] = shape->nodes[n_i][2];
```

Comment:
This code is copy pasted everywhere. Please refactor it in a function if you have teh same code twice.